website/content/docs/ecs/get-started/migrate-existing-tasks.mdx

```diff
@@ -0,0 +1,110 @@
+  requires_compatibilities = ["FARGATE"]
+  network_mode             = "awsvpc"
+  cpu                      = 256
+  memory                   = 512
+  execution_role_arn       = "arn:aws:iam::111111111111:role/execution-role"
+  task_role_arn            = "arn:aws:iam::111111111111:role/task-role"
+  container_definitions = jsonencode(
+    [{
+      name             = "example-client-app"
+      image            = "docker.io/org/my_task:v0.0.1"
+      essential        = true
+      portMappings = [
+        {
+          containerPort = 9090
+          hostPort      = 9090
+          protocol      = "tcp"
+        }
+      ]
+      cpu         = 0
+      mountPoints = []
+      volumesFrom = []
+    }]
+  )
+}
+
+resource "aws_ecs_service" "my_task" {
+  name            = "my_task"
+  cluster         = "arn:aws:ecs:us-east-1:111111111111:cluster/my-cluster"
+  task_definition = aws_ecs_task_definition.my_task.arn
+  desired_count   = 1
+  network_configuration {
+    subnets = ["subnet-abc123"]
+  }
+  launch_type            = "FARGATE"
+}
+```
+
+## Convert to the `mesh-task` Module
+
+In order to add the necessary sidecar containers for your task to join the mesh,
+you must use the [`mesh-task` module](https://registry.terraform.io/modules/hashicorp/consul-ecs/aws/latest/submodules/mesh-task).
+
+The `mesh-task` module uses inputs similar to your old ECS task definition but
+creates a new version of the task definition with additional containers.
+
+The `mesh-task` module is used as follows:
+
+```hcl
+module "my_task" {
+  source  = "hashicorp/consul/aws-ecs//modules/mesh-task"
+  version = "<latest version>"
+
+  family                = "my_task"
+  container_definitions = [
+    {
+      name         = "example-client-app"
+      image        = "docker.io/org/my_task:v0.0.1"
+      essential    = true
+      portMappings = [
+        {
+          containerPort = 9090
+          hostPort      = 9090
+          protocol      = "tcp"
+        }
+      ]
+      cpu         = 0
+      mountPoints = []
+      volumesFrom = []
+    }
+  ]
+
+  port       = "9090"
+  retry_join = "<address of the Consul server>"
+}
+```
+
+The main differences are:
+
+- You must remove the `execution_role_arn` and `task_role_arn` fields. The `mesh-task` module will create the task and execution roles.
+- You must add the `port` field. This is the port that your application listens on.
+- You must add the `retry_join` field. This specifies the location of your Consul servers so that your task can join the mesh.
+- You must remove the `jsonencode()` function from the `container_definitions` field.
+
+The `mesh-task` module will create a new version of your task definition with the
+necessary sidecar containers added so you can delete your existing `aws_ecs_task_definition`
+resource.
+
+Now that your task(s) are migrated to the `mesh-task` module, see the [Installation Guide](/docs/ecs/get-started/install) to continue installing
```

Comment:
Should this link to the routing subsection heading since they've already done the first part of that page?

Reply:
Since I didn't mention the ECS service here, I created a new `ECS Service` section in the Installation guide and linked to that instead.

I also added a `Next Steps` section here, and included a link to the mesh-task inputs reference.

Reply:
ahh gotcha. Sounds good.